### PR TITLE
add rADQLQC

### PR DESCRIPTION
**Pull request by sob2021**
# Pull Request

Adding rADQLQC.
Need package rgdsr

### CODE REVIEW

File: R/radqlqc.R [hunk 1]
```diff
@@ -0,0 +1,1304 @@
...
+#' @param cached boolean whether the cached <%= toupper(data) %> data \code{c<%=data%>} should be
+#' returned or new data
+#'
+#' @return a dataframe with SDTM questionnaire data
+#'
+#' @import random.cdisc.data
+#' @import dplyr
+#' @importFrom rgdsr gdsr_init sdtmv_variables
+#'
+#' @export
```
Comment (shajoezhu) [inline R/radqlqc.R:291]:
```suggestion
```

We can make this function as internal. no need to export

File: R/radqlqc.R [hunk 2]
```diff
@@ -0,0 +1,1304 @@
...
+    ) %>%
+    select(-.data$publication_name)
+
+  # ADaM global health status and scales from gdsr
+  gdsr_param_adqlqc <- adamv_parameters()[["ADQLQC"]] %>%
+    select(
+      .data$PARAM,
+      .data$PARAMCD,
+      .data$PARCAT2,
+      .data$PARCAT1N
```
Comment (shajoezhu) [inline R/radqlqc.R:830]: Can we extract the dataframe from rgdsr, and just keep these columns and save it as dataset, and load it in?

This is repeated multiple times. Note, we are not updating this, so any version is fine

File: R/radqlqc.R [hunk 3]
```diff
@@ -0,0 +1,1304 @@
...
+  )
+  adsl_vars <- adsl_vars$name
+
+  # adsl variables from ADQLQC in gdsr
+  adsl_vars <- filter(
+    adamv_variables(),
+    .data$dataset == "ADQLQC",
+    str_detect(.data$derivation_text, "^ADSL")
+  )
+  adsl_vars <- adsl_vars$name
```
Comment (shajoezhu) [inline R/radqlqc.R:145]: duplicated. for removal

File: R/radqlqc.R [hunk 4]
```diff
@@ -0,0 +1,1304 @@
...
+  )
+
+  # Merge ADSL --------------------------------------------------------------
+  # adsl variables from ADQLQC in gdsr
+  adsl_vars <- filter(
+    adamv_variables(),
+    .data$dataset == "ADQLQC",
+    str_detect(.data$derivation_text, "^ADSL")
+  )
+  adsl_vars <- adsl_vars$name
```
Comment (shajoezhu) [inline R/radqlqc.R:137]: Can we just hardcode this list as 

 [1] "STUDYID"  "USUBJID"  "SUBJID"   "SITEID"   "REGION1"  "COUNTRY"  "ETHNIC"   "AGE"     
 [9] "AGEU"     "AAGE"     "AAGEU"    "AGEGR1"   "AGEGR2"   "AGEGR3"   "STRATwNM" "STRATw"  
[17] "STRATwV"  "SEX"      "RACE"     "ITTFL"    "SAFFL"    "PPROTFL"  "TRT01P"   "TRTxxP"  
[25] "TRT01A"   "TRTxxA"   "TRTSEQP"  "TRTSEQA"  "TRTSDTM"  "TRTSDT"   "TRTEDTM"  "TRTEDT"  
[33] "DCUTDT"

**Review by shajoezhu — CHANGES_REQUESTED** [review]
Hi @sob2021 , many thanks for your changes, can we please remove the useage of rgdsr, some of the variables names, we can just hard-coded. 

for that dataset I think we can save dataframe and load it. Thanks a lot!

File: R/radqlqc.R (hunk 1 above)
Comment (sob2021) [reply R/radqlqc.R:291]: Thank you @shajoezhu, export removed

File: R/radqlqc.R (hunk 2 above)
Comment (sob2021) [reply R/radqlqc.R:830]: Thank you, functions and code for rgdsr removed. Generated data stored in inst/data.

File: R/radqlqc.R (hunk 3 above)
Comment (sob2021) [reply R/radqlqc.R:145]: Thank you, removed.

File: R/radqlqc.R (hunk 4 above)
Comment (sob2021) [reply R/radqlqc.R:137]: Thank you, variables now hardcoded.

**Comment by shajoezhu** [comment]
Thanks a lot @sob2021 for working on these issues, once the code is ready, you can run the code in the vignete/build_cached_data to compile the data, git add them and commit as well. Thanks a lot

**Comment by sob2021** [comment]
Hi @shajoezhu 
I've made all the updates as discussed.

Thank you.

**Comment by shajoezhu** [comment]
Hi @sob2021 , after removing that N parameter, you will need to recompile the dataset.

**Pull request merged**